### PR TITLE
build: drop SWC; use TS for cjs/esm bundles

### PR DESCRIPTION
**Pull request by dimitrisnl**
### What changed

- Drop SWC in favour of using just TypeScript
- Introduce 3 different tsconfigs. 1) ESM, 2) CJS, 3) Types
- Completely drop `exports` key

### Other References
- [FP-TS](https://github.com/gcanti/fp-ts/blob/master/package.json#L5)
- [sum-types by unsplash](https://github.com/unsplash/sum-types/blob/master/package.json#L22)

### Beta link
https://www.npmjs.com/package/zod-validation-error/v/0.2.2-beta.1

### CODE REVIEW

**Comment by dimitrisnl** [comment]
ESM seems to be working

```sh
dnl::src ⌁ node main.mjs                             
ValidationError: Validation error: Required at "name"
    at fromZodError (<project-path>/node_modules/zod-validation-error/dist/cjs/ValidationError.js:27:12)
    at parseUser (file://<project-path>/src/main.mjs:14:29)
    at file://<project-path>/src/main.mjs:21:1
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
    at async loadESM (node:internal/process/esm_loader:91:5)
    at async handleMainPromise (node:internal/modules/run_main:65:12) {
  details: [
    {
      code: 'invalid_type',
      expected: 'string',
      received: 'undefined',
      path: [Array],
      message: 'Required'
    }
  ]
}
```

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (dimitrisnl) [reply jest.config.js:1]: Frankly, I didn't like that it didn't have the jest icon in VSCode.

File: package.json [hunk 2]
```diff
@@ -87,18 +78,15 @@
...
+    "ts-jest": "^29.0.3",
+    "typescript": "^4.9.4",
+    "zod": "^3.20.0"
   },
   "peerDependencies": {
-    "zod": "^3.18.0"
-  },
-  "dependencies": {
-    "@swc/helpers": "^0.4.11"
+    "zod": "^3.20.0"
```
[the remark this answers is not in the document]
Comment (dimitrisnl) [reply package.json:90]: Hm, I incorrectly assumed it was a devDependency. Let me check

File: jest.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dimitrisnl) [reply jest.config.js:1]: Also most importantly there was an odd issue when upgrading to TS4.9 that was resolved when changing the file to a simple js one.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (dimitrisnl) [reply package.json:90]: I don't think there will be an error. We're using the same codebase for both 18 & 20. It's a minor bump too. I can't see how a user-side error can occur

[the remark this answers is not in the document]
Comment (dimitrisnl) [reply package.json:90]: They only think that can happen is having a warning of `zod-validation-error has an unmeet peerDependency zod>=3.20`

[the remark this answers is not in the document]
Comment (jmike) [reply package.json:90]: Isn't this a problem? What if we left this line unchanged?

[the remark this answers is not in the document]
Comment (dimitrisnl) [reply package.json:90]: You're right, let's not change it.

**Comment by jmike** [comment]
@nikoskalogridis can you also afford some time to review this please?

**Pull request merged**